glean-core/src/ac_migration/mod.rs

```diff
@@ -0,0 +1,108 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! A module containing glean-core code for supporting data migration
+//! from glean-ac.
+
+use crate::util::truncate_string_at_boundary;
+use std::collections::HashMap;
+
+use super::Glean;
+use super::PingMaker;
+
+const GLEAN_AC_SEQUENCE_SUFFIX: &str = "_seq";
+
+/// The Glean A-C metadata migration object.
+#[derive(Debug, Clone)]
+pub struct ACMetadata {
+    /// A map in which entries are (`<ping_name>_seq`, nextSequenceNumber)..
+    pub sequence_numbers: HashMap<String, i32>
+}
+
+/// Stores the sequence numbers from glean-ac in glean-core.
+pub(super) fn migrate_sequence_numbers(
+    glean: &Glean,
+    seq_numbers: HashMap<String, i32>
+) {
+    let ping_maker = PingMaker::new();
+
+    for (store_name_with_suffix, next_seq) in seq_numbers.iter() {
+        // Note: glean-ac stores the sequence numbers as '<ping_name>_seq',
+        // glean-core requires '<ping_name>#sequence'.
```

Comment:
I guess we could change that to `_seq` as well if we want?
We'd still need a check anyway I guess, so just keeping it is just as good.

Reply:
Yes, I thought about doing that, but then I realized we'd still need a check. So I stayed with this. Happy to change though, depending on other's feedback.

Reply:
I prefer the `#sequence` going forward, since `#` feels like something we reserve for an internal detail, whereas underscores appear in ping names routinely...